crates/holochain_sqlite/tests/tests/peer_meta_store.rs

```diff
@@ -0,0 +1,253 @@
+use holochain_sqlite::db::{DbKindPeerMetaStore, DbWrite, ReadAccess};
+use holochain_sqlite::error::DatabaseResult;
+use holochain_sqlite::sql::sql_peer_meta_store::{DELETE, GET, INSERT, PRUNE};
+use rusqlite::params;
+use std::sync::Arc;
+
+#[tokio::test]
+async fn insert_read_delete() {
+    let peer_meta_store = DbWrite::test_in_mem(DbKindPeerMetaStore(Arc::new(
+        kitsune2_api::SpaceId::from(bytes::Bytes::from_static("test".as_bytes())),
+    )))
+    .unwrap();
+
+    let peer_url = kitsune2_api::Url::from_str("ws://test:80/1").unwrap();
+    peer_meta_store
+        .write_async({
+            let peer_url = peer_url.clone();
+            move |txn| -> DatabaseResult<()> {
+                txn.execute(
+                    INSERT,
+                    params! {
+                        peer_url.as_str(),
+                        "test",
+                        "test-value".as_bytes(),
+                        None::<u32>,
+                    },
+                )?;
+
+                Ok(())
+            }
+        })
+        .await
+        .unwrap();
+
+    let value = peer_meta_store
+        .read_async({
+            let peer_url = peer_url.clone();
+            move |txn| -> DatabaseResult<Vec<u8>> {
+                let value = txn.query_row(
+                    GET,
+                    params! {
+                        peer_url.as_str(),
+                            "test",
+                    },
+                    |row| row.get(0),
+                )?;
+
+                Ok(value)
+            }
+        })
+        .await
+        .unwrap();
+
+    assert_eq!("test-value".as_bytes(), value);
+
+    peer_meta_store
+        .read_async({
+            let peer_url = peer_url.clone();
+            move |txn| -> DatabaseResult<()> {
+                txn.execute(
+                    DELETE,
+                    params! {
+                        peer_url.as_str(),
+                            "test",
+                    },
+                )?;
+
+                Ok(())
+            }
+        })
+        .await
+        .unwrap();
+
+    let row_count = peer_meta_store
+        .read_async(move |txn| -> DatabaseResult<u32> {
+            let row_count = txn.query_row("SELECT COUNT(*) FROM peer_meta", [], |row| {
+                row.get::<_, u32>(0)
+            })?;
+
+            Ok(row_count)
+        })
+        .await
+        .unwrap();
+
+    assert_eq!(1, row_count);
+}
+
+#[tokio::test]
+async fn prune() {
+    let peer_meta_store = DbWrite::test_in_mem(DbKindPeerMetaStore(Arc::new(
+        kitsune2_api::SpaceId::from(bytes::Bytes::from_static("test".as_bytes())),
+    )))
+    .unwrap();
+
+    let peer_url = kitsune2_api::Url::from_str("ws://test:80/1").unwrap();
```

Comment:
This would be a good addition to the kitsune2 test utils.

Reply:
Yes, again if we want to have that as a dependency here. It's currently unpublished and expected to be used within K2

Reply:
Even just for kitsune.